pymc3/distributions/timeseries.py

```diff
@@ -309,8 +311,14 @@ def _random(self, sigma, mu, size, sample_shape):
         else:
             axis = 0
         rv = stats.norm(mu, sigma)
-        data = rv.rvs(size).cumsum(axis=axis)
-        data = data - data[0]  # TODO: this should be a draw from `init`, if available
+        if len(size) == 1:
+            data = rv.rvs(size).cumsum(axis=axis)
+            data = data - data[0]  # TODO: this should be a draw from `init`, if available
+        else:
+            data = np.empty(size)
+            for i in range(size[0]):
+                data[i]=rv.rvs((size[1],)).cumsum(axis=axis) 
+                data[i]=data[i] - data[i][0]
```

Comment:
I don't think this generalizes well to cases where the dimensions of `size` are 0 or > 2. This would explain why [tests are failing](https://travis-ci.org/github/pymc-devs/pymc3/jobs/703452672#L3104) with broadcast errors like `ValueError: could not broadcast input array from shape (5) into shape (5,10)`

Reply:
Instead something like
```
for d, s in enumerate(size):
    ...
```
?

And this may be a bit more efficient:
```
data[i] -= data[i, 0]
```

Reply:
This is what I did. However, I would incorporate your advice `data[i] -= data[i, 0]`
`            data = np.empty(size)
            list_of_size = list(size)
            size_inner_matrix = list_of_size[1:]
            size_inner_matrix_tuple = tuple(size_inner_matrix)
            for i in range(size[0]):
                data[i]=rv.rvs(size_inner_matrix_tuple).cumsum(axis=axis) 
                data[i]=data[i] - data[i][0]`

Reply:
I did it in this way:
`            data = np.empty(size)<br>
            list_of_size = list(size)
            size_inner_matrix = list_of_size[1:]
            size_inner_matrix_tuple = tuple(size_inner_matrix)
            for i in range(size[0]):
                data[i]=rv.rvs(size_inner_matrix_tuple).cumsum(axis=axis) 
                data[i]=data[i] - data[i][0]`

Reply:
`            data = np.empty(size)
                list_of_size = list(size)
                size_inner_matrix = list_of_size[1:]
                size_inner_matrix_tuple = tuple(size_inner_matrix)
                for i in range(size[0]):
                    data[i]=rv.rvs(size_inner_matrix_tuple).cumsum(axis=axis) 
                    data[i]=data[i] - data[i][0]`

Reply:
`            data = np.empty(size)/
            list_of_size = list(size)/
            size_inner_matrix = list_of_size[1:]/
            size_inner_matrix_tuple = tuple(size_inner_matrix)/
            for i in range(size[0]):/
                data[i]=rv.rvs(size_inner_matrix_tuple).cumsum(axis=axis)/ 
                data[i]=data[i] - data[i][0]/`

Reply:
This is how I did it.

```
            data = np.empty(size)
            list_of_size = list(size)
            size_inner_matrix = list_of_size[1:]
            size_inner_matrix_tuple = tuple(size_inner_matrix)
            for i in range(size[0]):
                data[i]=rv.rvs(size_inner_matrix_tuple).cumsum(axis=axis) 
                data[i]=data[i] - data[i][0]
```

Reply:
Thanks for the advice. Apart from the broadcast errors, there is one `ImportError` of `test_kronecker_normal `which does not have any bearing on my code. What should I do about it?

Also, what do I need to do with `xfailed `and `skipped `tests?

Reply:
@AlexAndorra @michaelosthege Do you have any idea why this particular [test case is failing?](https://travis-ci.org/github/pymc-devs/pymc3/jobs/704187938)

Reply:
It seems like your implementation still has difficulties broadcasting the shape when there is only one dimension, i.e going from (5,) to (1, 5) or (5, 1) 🤔 The best would probably be to reimplement the test case in your local environment, and try to see where things go wrong -- that's usually what I do.

More generally, if you feel like it, it'd be great to fill an issue on Scipy's repo too, because in the end the problem seems to come from their side. Once fixed, we wouldn't need this workaround anymore.

Reply:
Yeah, I have raised an issue on Scipy's repo [here](https://github.com/scipy/scipy/issues/12482).
Also, I have fixed the implementation for the only remaining failed test case.